anagram failing to compile on Linux Ubuntu 14.04

Apparently the tolower and isalpha macros involve a function call that returns an unchecked pointer.  Here is an example error:
```
mnt/d/autobahn1/llvm-test-suite/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c:300:17: error: cannot use a return with an unchecked type in a checked scope or function
            if (isalpha(ch)) cLetters++;
                ^
/usr/include/ctype.h:198:21: note: expanded from macro 'isalpha'
# define isalpha(c)     __isctype((c), _ISalpha)
                        ^
/usr/include/ctype.h:89:6: note: expanded from macro '__isctype'
  ((*__ctype_b_loc ())[(int) (c)] & (unsigned short int) type)
     ^
```

--- a/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
+++ b/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
@@ -259,6 +259,8 @@ _Unchecked void Fatal(char *pchMsg, unsigned u) {
 
 int isalpha_checked(int ch) _Unchecked { return isalpha(ch); }
 #define isalpha(ch) isalpha_checked(ch)
+int tolower_checked(int ch) _Unchecked { return tolower(ch); }
+#define tolower(ch) tolower_checked(ch)
 
 /* ReadDict -- read the dictionary file into memory and preprocess it
  *

--- a/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
+++ b/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
@@ -257,6 +257,9 @@ _Unchecked void Fatal(char *pchMsg, unsigned u) {
     exit(1);
 }
 
+int isalpha_checked(int ch) _Unchecked { return isalpha(ch); }
+#define isalpha(ch) isalpha_checked(ch)
+
 /* ReadDict -- read the dictionary file into memory and preprocess it
  *
  * A word of length cch in the dictionary is encoded as follows:

--- a/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
+++ b/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
@@ -257,12 +257,8 @@ _Unchecked void Fatal(char *pchMsg, unsigned u) {
     exit(1);
 }
 
-#ifdef isalpha
-# undef isalpha
-#endif
-#ifdef tolower
-# undef tolower
-#endif
+#undef isalpha
+#undef tolower
 
 /* ReadDict -- read the dictionary file into memory and preprocess it
  *

--- a/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
+++ b/MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
@@ -257,10 +257,12 @@ _Unchecked void Fatal(char *pchMsg, unsigned u) {
     exit(1);
 }
 
-int isalpha_checked(int ch) _Unchecked { return isalpha(ch); }
-#define isalpha(ch) isalpha_checked(ch)
-int tolower_checked(int ch) _Unchecked { return tolower(ch); }
-#define tolower(ch) tolower_checked(ch)
+#ifdef isalpha
+# undef isalpha
+#endif
+#ifdef tolower
+# undef tolower
+#endif
 
 /* ReadDict -- read the dictionary file into memory and preprocess it
  *

